internal/api/webhook_receivers.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+// WebhookReceiverType is an enum representing the type of a webhook receiver.
+// It is used to identify the platform or service that the webhook receiver is
+// associated with, such as GitHub or Quay.
+type WebhookReceiverType string
```

Comment:
This (and its values) really ought to stay in `project_config_types.go` so they can be annotated similarly to here to aid with validation and generated docs:

https://github.com/akuity/kargo/blob/b031b380a32bdc01d09f80d224f75a2b52e35f60/api/v1alpha1/warehouse_types.go#L15-L16

Reply:
The keys for the data maps can stay put.